File: Solutions/1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> mapIndex = new HashMap<>();
        for(int i=0;i<nums.length;++i){
            mapIndex.put(nums[i],i);
        }
        for(int i=0;i<nums.length;++i){
            int temp=target-nums[i];
            if(mapIndex.containsKey(temp) && mapIndex.get(temp) != i){
                return new int[]{i,mapIndex.get(temp)};
            }
        }
        throw new IllegalArgumentException("No two sum solution");
    }
}

File: Solutions/1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let mapIndex = new Map();
    let len = nums.length;
    for(let i=0;i<len;i++){
        mapIndex.set(nums[i],i);
    }
    for(let i=0;i<len;i++){
        let temp=target-nums[i];
        if(mapIndex.has(temp) && mapIndex.get(temp) != i){
            return [i,mapIndex.get(temp)];
        }
    }
};